Source/NSHost.m

```diff
@@ -530,6 +530,19 @@ + (NSHost*) hostWithAddress: (NSString*)address
 #if     defined(AF_INET6)
     {
       struct in6_addr	hostaddr6;
+      
+      //Remove brackets if exists
+      const char *formattedAddress;
+      NSUInteger addressLength = [address length] - 1; //Don't fall off the array
+      char firstChar = [address characterAtIndex:0];
+      char lastChar = [address characterAtIndex:addressLength];
+
+      if(firstChar == '[' && lastChar == ']')
+      {
+          NSString *lastBracketRemoved = [address substringToIndex:addressLength];
+          NSString *cleanedAddress = [lastBracketRemoved substringFromIndex:1];
+          a = [cleanedAddress UTF8String];
+      }
```

Comment:
what happens if this is not the case? Shouldn't we do something here?

Reply:
If that's not the case then it's an IPv4 address. If for some miracle an invalid address made it here then it will fail in the proceeding checks.